services/director-v2/src/simcore_service_director_v2/modules/osparc_variables/substitutions.py

```diff
@@ -137,6 +138,37 @@ def create(cls, app: FastAPI):
         return table
 
 
+_NEW_ENVIRONMENTS: Final = {
+    "OSPARC_API_BASE_URL": "$OSPARC_VARIABLE_API_HOST",
+    "OSPARC_API_KEY": "$OSPARC_VARIABLE_API_KEY",
+    "OSPARC_API_SECRET": "$OSPARC_VARIABLE_API_SECRET",
+    "OSPARC_STUDY_ID": "$OSPARC_VARIABLE_STUDY_UUID",
+    "OSPARC_NODE_ID": "$OSPARC_VARIABLE_NODE_ID",
+}
+
+
+def auto_inject_environments(
+    compose_spec: ComposeSpecLabelDict,
+) -> ComposeSpecLabelDict:
+    # SEE https://github.com/ITISFoundation/osparc-simcore/issues/5925
+    for service in compose_spec.get("services", {}).values():
+        current_environment = deepcopy(service.get("environment", {}))
+
+        # if _NEW_ENVIRONMENTS are already defined, then do not change them
+        if isinstance(current_environment, dict):
+            service["environment"] = {
+                **_NEW_ENVIRONMENTS,
+                **current_environment,
+            }
+        elif isinstance(current_environment, list):
+            service["environment"] += [
+                f"{name}={value}"
+                for name, value in _NEW_ENVIRONMENTS.items()
+                if not any(_.startswith(name) for _ in current_environment)
```

Comment:
I will officially complain about this. By convention `_` means it's not used. This confuses me, please use a different identifier.

Reply:
ok. 
I saw this in other repos and found it very handy to remove "noise" variables from the code. 
IMO it reads better but if it confuses you, i can use some small identifier